panel/models/html.ts

```diff
@@ -127,12 +141,27 @@ export class HTMLView extends PanelMarkupView {
   }
 
   set_html(html: string | null): void {
-    if (html !== null) {
-      this.container.innerHTML = html
-      if (this.model.run_scripts) {
-        run_scripts(this.container)
+    if (html === null) {
+      return
+    }
+    this.container.innerHTML = html
+    if (this.model.run_scripts) {
+      run_scripts(this.container)
+    }
+    this._setup_event_listeners()
+    for (const anchor of this.container.querySelectorAll("a")) {
+      const link = anchor.getAttribute("href")
+      if (link && link.startsWith("#")) {
+        anchor.addEventListener("click", () => {
+          const found = searchAllDOMs(document.body, link)
+          if ((found.length > 0) && found[0] instanceof Element) {
+            found[0].scrollIntoView()
+          }
+        })
+        if (!this.root.has_finished() && this.model.document && window.location.hash === link) {
+          this.model.document.on_event("document_ready", () => anchor.scrollIntoView())
```

Comment:
```suggestion
          this.model.document.on_event("document_ready", () => {
            anchor.scrollIntoView()
            setTimeout(() => anchor.scrollIntoView(), 5)
          })
```

I need to wait for the document to load; without this change, the original issue looks like this. (I'm running it twice so we are closer to the anchor for the second run.)

![image](https://github.com/user-attachments/assets/318c6ece-72b9-4ee1-96f9-eb273edcf65a)

Reply:
May be have a test for this too.

Reply:
Will try with `requestAnimationFrame` instead.

Reply:
Wasn't enough, switched to the timeout.